src/test/java/seedu/address/model/appointment/PastAppointmentTest.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void parseLocalDate_invalidDate_throwsDateTimeParseException() {
+        assertThrows(DateTimeParseException.class, () -> PastAppointment.parseLocalDate("01-01-2020 12:00"));
+    }
+
+    @Test
+    public void parseLocalDate_pastDate_returnsLocalDate() {
+        assertEquals(LocalDate.parse("01-01-2020", DateTimeFormatter.ofPattern(DATE_FORMAT)),
+                PastAppointment.parseLocalDate("01-01-2020"));
+    }
+
+    @Test
+    public void parseLocalDate_today_returnsLocalDate() {
+        assertEquals(LocalDate.now(), PastAppointment.parseLocalDate(LocalDate.now()
+                .format(DateTimeFormatter.ofPattern(DATE_FORMAT))));
+    }
+
+    @Test
+    public void getMedication_validMedication_returnsMedication() {
+        assertEquals("[[paracetamol]]", new PastAppointmentBuilder()
+                .withMedication(new String[] { "Paracetamol" }).build().getMedication().toString());
+    }
+
+    @Test
+    public void getMedication_nullMedication_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new PastAppointmentBuilder().withMedication(null).build());
+    }
+
+    @Test
+    public void getMedication_emptyMedication_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> new PastAppointmentBuilder()
+                .withMedication(new String[] { "" }).build());
+    }
+
+    @Test
+    public void getMedication_multipleMedication_returnsMedication() {
+        assertEquals("[[ibuprofen], [paracetamol]]", new PastAppointmentBuilder()
+                .withMedication(new String[] { "Paracetamol", "Ibuprofen" }).build().getMedication().toString());
+    }
+
+    @Test
+    public void getMedication_multipleMedicationWithEmpty_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> new PastAppointmentBuilder()
+                .withMedication(new String[] { "Paracetamol", "" }).build());
+    }
+
+    @Test
+    public void getDiagnosis_validDiagnosis_returnsDiagnosis() {
+        assertEquals("Fever", new PastAppointmentBuilder()
+                .withDiagnosis("Fever").build().getDiagnosis());
+    }
+
+    @Test
+    public void getMedicationString_validMedication_returnsMedicationString() {
+        assertEquals("Medication: paracetamol", new PastAppointmentBuilder()
+                .withMedication(new String[] { "Paracetamol" }).build().getMedicationString());
+    }
+
+    @Test
+    public void getMedicationString_multipleMedication_returnsMedicationString() {
+        assertEquals("Medication: ibuprofen, paracetamol", new PastAppointmentBuilder()
+                .withMedication(new String[] { "Paracetamol", "Ibuprofen" }).build().getMedicationString());
+    }
+
+    @Test
+    public void equals() {
+        PastAppointment pastAppointment;
+
+        pastAppointment = new PastAppointmentBuilder().build();
+        assertNotEquals(null, pastAppointment); // different type
```

Comment:
Not a big issue here, but to follow our code formatting, I believe it was in the order:
Comment -> Construct necessary variables -> Assert

instead of your order of Variable -> Assert -> Comment.

Reply:
Done!